### PR TITLE
Imrovements to api-review-process.md

### PR DESCRIPTION
**Pull request by SingleAccretion**
This is the result of @danmosemsft's comment [here](https://github.com/dotnet/runtime/issues/38160#issuecomment-649063068).

This PR adds three links to the "API Review Process" document:
 - A link to the template for an API proposal: as an addition to "a good issue" example.
 - A link to the YouTube playlist with API reviews: to mention that the process of review is actually public.
 - http://aka.ms/ready-for-api-review: as a useful link for people interested in API reviews.

I also initially thought of adding a link to [apisof.net](https://apisof.net/), but after exploring this possibility decided against it due to the lack of a good place for it.

### CODE REVIEW

File: docs/project/api-review-process.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ The rule of thumb is that we (**dotnet/runtime**) review every API that is being
 
 ## Steps
 
-1. **Requester files an issue**. The issue description should contain a speclet that represents a sketch of the new APIs, including samples on how the APIs are being used. The goal isn't to get a complete API list, but a good handle on how the new APIs would roughly look like and in what scenarios they are being used. Here is [a good example](https://github.com/dotnet/corefx/issues/271).
+1. **Requester files an issue**. The issue description should contain a speclet that represents a sketch of the new APIs, including samples on how the APIs are being used. The goal isn't to get a complete API list, but a good handle on how the new APIs would roughly look like and in what scenarios they are being used. Here is [a good example](https://github.com/dotnet/corefx/issues/271), you can also use [the template](https://github.com/dotnet/runtime/issues/new?assignees=&labels=api-suggestion&template=02_api_proposal.md&title=).
```
Comment (danmosemsft) [inline docs/project/api-review-process.md:15]:
```suggestion
1. **Requester files an issue**. The issue description should contain a speclet that represents a sketch of the new APIs, including samples on how the APIs are being used. The goal isn't to get a complete API list, but a good handle on how the new APIs would roughly look like and in what scenarios they are being used. Please use [this template](https://github.com/dotnet/runtime/issues/new?assignees=&labels=api-suggestion&template=02_api_proposal.md&title=). The issue should have the label `api-suggestion`. Here is [a good example](https://github.com/dotnet/runtime/issues/38344) of an issue following that template.
```

**Comment by danmosemsft** [comment]
Thanks @SingleAccretion I pushed some other changes - do they look OK? Better to use a more modern example so I updated that.

Ideally this would all be written with active voice ("you" instead of "the requester") to be easier to read.

**Comment by SingleAccretion** [comment]
@danmosemsft Yes, I very much agree with the changes. 

The issue you mentioned kind of goes deeper in my opinion (I actually had a little write up about this at first in the PR description, but later cut it out for brevity). The document addresses two kinds of people at once, and because of that cannot really use "you" effectively, since in the first half, "you" would mean "the requester", but in the second half, it would be "the maintainer", someone who is looking for guidance on how the review process works from their perspective. 

I personally think this is fine, since having two documents/two sections that describe basically the same process would (in my opinion) be worse than the tradeoff of using a lot of passive voice.

**Review by danmoseley — APPROVED** [review]

**Comment by danmosemsft** [comment]
@SingleAccretion sounds fine, I've merged as is (thank you) and if you feel like any future improvements just throw up a new PR.

**Pull request merged**